### DISCUSSION TITLE
抖音上的數學

### DISCUSSION THREAD

**Discussion by gigitilto** [1101-GS4719-C/Python-Community-GS4719-C#2248] [General]
昨天看到抖音一個很唬爛的算數，如圖，
<img width="252" alt="截圖 2024-11-06 下午2 00 40" src="https://github.com/user-attachments/assets/9afcc691-e427-4df2-a7c0-02d4ff25f644">
將根號內的數字相加，扣除開根號的2，竟然可以得到答案，但很顯然只有這四個數字是符合的，留言區罵聲一片，我卻很好奇是否在更高階也存在這樣的數字？於是我寫了一個python程式去計算在開k次方根下符合這個算法規則的數字，令人驚訝的是，確實是存在的，以開200次方根為例，最大的這個數是2972^200，將這個數的所有位數相加減200剛好就是2972。雖然我不會證明k趨近無限時也會存在這樣的數，但我從k在1~200的圖可以發現一些規則，如圖，
![res3](https://github.com/user-attachments/assets/f6317b88-ec40-4d91-9c42-e2323c50bf05)
不是所有的k值都會有這樣的數字，但這些k值都會是2或3的倍數，而且最大的數也有越來越大的趨勢，但並非單調遞增。

**Comment by Chattenoire** [comment]
你可以把數字限定為質數，看看會不會有這樣的分布；如果行有餘力，可以證明這個到無限都會存在規律，就可以算是個不錯的小project；至於要generalize的話，可以試試看N次方根或是其他方向。
